runtime/lua/vim/fs.lua

```diff
@@ -679,4 +679,22 @@ function M.rm(path, opts)
   end
 end
 
+--- Returns `true` if `path` is relative to `root`
+--- @param path string?
+--- @param root string?
+--- @return boolean
+function M.is_relative_to(path, root)
+  if root == nil or path == nil then
```

Comment:
if they are both optional then we should return `true` if they are both nil.

not sure if they should be optional though. callers can opt-in to "relaxed" handling via `path or ''`.

Reply:
No strong opinions on this, string-only sounds good.